internal/configs/named_values.go

```diff
@@ -110,6 +115,15 @@ func decodeVariableBlock(block *hcl.Block, override bool) (*Variable, hcl.Diagno
 		v.SensitiveSet = true
 	}
 
+	if attr, exists := content.Attributes["nullable"]; exists {
+		valDiags := gohcl.DecodeExpression(attr.Expr, nil, &v.Nullable)
+		diags = append(diags, valDiags...)
+	} else {
+		// The current default is true, which is subject to change in a future
+		// language edition.
+		v.Nullable = true
+	}
+
```

Comment:
Do we need to do anything special here to deal with override files?

I don't know if there's a real use-case for overriding the "nullability" of a variable, but if we want to support it I think we'd need to have a `NullableSet` field alongside so that the override-merging logic can understand whether the override file is setting it or not.

Alternatively, we could decide that override files are not allowed to change nullability -- it doesn't really seem to be the sort of thing the override files feature is normally used for -- and in that case just return an error in the `exists` case if the `override` flag is also set, effectively making it invalid to set `nullable` in override files.

I initially thought we'd need to allow `nullable = true` in override files so that a future edition upgrade could add it there, but I think that's not actually true because we only allow override files to change the definition of variables already declared and never to declare entirely new variables, and so we'd always be able to add the `nullable = true` to the original declaration and get the same effect as adding it to the override file, assuming we forbid setting it in override files.

Reply:
Thanks, I did forget to prevent this from being overridden. I don't see a compelling reason to allow overriding `nullable`, or at least we can start more conservatively and allow it later if it would prove to be useful.

Reply:
Actually since we make the nullable value true by default during decoding, and we can override the `Default` value, I had to handle this in both in decoding and merging anyway. Since the merging rules are clearly laid out already I went with them for now, and allowed the explicit overriding of `nullable`, however the implicit `true` setting will not override a given block.